src/arrayMethodJoin.js

```diff
@@ -4,8 +4,24 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let stringFromArray = '';
+
+    const endOfArray = this.length - 1;
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] === undefined || this[i] === null) {
+        stringFromArray += '';
+      } else {
+        stringFromArray += this[i];
+      }
```

Comment:
No need in else statement. Because you add nothing.
```suggestion
      if (this[i] !== undefined || this[i] !== null) {
        stringFromArray += this[i];
      }
```